cmd/kubeadm/app/preflight/checks.go

```diff
@@ -987,7 +990,7 @@ func addCommonChecks(execer utilsexec.Interface, cfg kubeadmapi.CommonConfigurat
 		IsPrivilegedUserCheck{},
 		HostnameCheck{nodeName: cfg.GetNodeName()},
 		KubeletVersionCheck{KubernetesVersion: cfg.GetKubernetesVersion(), exec: execer},
-		ServiceCheck{Service: "kubelet", CheckIfActive: false},
+		ServiceCheck{Service: "kubelet", WantActive: false},
```

Comment:
This seems to change the behavior of the kubelet check significantly here...  It looks like previously CheckIfActive would skip the active check completely, but now with WantActive this is testing specifically for the kublet not being active.

Reply:
Yes, that's what I'm suggesting. The reasons are explained in the commit message.

Reply:
Right, but since we are checking for `active` or `activating` [here](https://github.com/kubernetes/kubernetes/blob/7b8bb6e7d3345b04678d96a721f651a4a68eb1dd/pkg/util/initsystem/initsystem.go#L78) wouldn't this preflight check fail by default on any system that has the kubelet enabled/started before running kubeadm (which is the default case on Ubuntu/Debian systems)?

Reply:
Here's my observation: if kubelet is ```active```, ```kubeadm init``` would fail in markmaster timeout (I was hit by this). If kubelet is ```activating```, it's actually crash-looping and can only start/recover once the configurations are created by ```kubeadm```. 

So I thought to make it more bullet proof and make sure we start with kubelet not running at all. That's the state where ```kubeadm reset``` leaves it too.

Reply:
@mythi that behavior makes perfect sense to me and is the default behavior when installing from the rpms. However the expected behavior with debs is that the service is enabled and started after installation, which is also the behavior with the current debs that we ship.

While I personally don't agree with the default behavior of enabling/starting daemons on package install, it is the expected behavior on Debian/Ubuntu systems.

That said, I believe the only reason that we are currently checking for `active` or `activating` is for the check in kubeadm.

Reply:
@detiber I'm not sure I got your point wrt. to my change proposal. I don't mind to keep kubelet running as it is today (and I don't mind if this PR is not OK) as long as the behaviour is consistent. Right now you have to have it _crash-looping_ in order to get ```kubeadm init``` to succeed. For newcomers like me it'll take some time to realize why ```active``` is not sufficient.

Reply:
@mythi maybe I'm not fully grokking these changes, but it appears to me that the preflight check would currently fail on an Ubuntu host because we are testing that initSystem.ServiceIsActive() returns false for the kubelet, since ServiceIsActive() is checking for 'active' or 'activating' and the kubelet service on Ubuntu will be in 'activating' by default.

Reply:
@detiber Currently the preflight check for kubelet ignores the activity status (```CheckIfActive``` is set to ```false```).